util/src/main/java/com/farao_community/farao/util/SystematicSensitivityAnalysisService.java

```diff
@@ -71,8 +87,35 @@ public static SensitivitySecurityAnalysisResult runSensitivity(Network network,
         }
         network.getVariantManager().setWorkingVariant(initialVariantId);
 
-        //return SensitivitySecurityAnalysisResult
-        return new SensitivitySecurityAnalysisResult(precontingencyResult, contingencySensitivityComputationResultsMap);
+        //return SystematicSensitivityAnalysisResult
+        return new SystematicSensitivityAnalysisResult(precontingencyResult, preMargin, contingencySensitivityComputationResultsMap, contingencyReferenceMarginMap);
+    }
+
+    private static void buildMarginFromNetwork(Network network, Crac crac, Map<String, Double> referenceMargin) {
+        Set<Cnec> cnecs = crac.getCnecs();
+        crac.synchronize(network);
+        for (Cnec cnec : cnecs) {
+            double margin = 0.0;
+            //get from network
+            String cnecnetworkelementid = cnec.getCriticalNetworkElement().getId();
+            Branch branch = network.getBranch(cnecnetworkelementid);
+            if (branch == null) {
+                LOGGER.error("Cannot found branch in network for cnec: {} during building reference flow from network.", cnecnetworkelementid);
+            } else {
+                try {
+                    margin = cnec.computeMargin(network);
+                } catch (SynchronizationException e) {
+                    //Hades config "hades2-default-parameters:" should be set to "dcMode: false"
+                    LOGGER.error("Cannot get compute margin for cnec {} in network variant.", cnecnetworkelementid);
+                } catch (FaraoException e) {
+                    LOGGER.error(e.getMessage());
+                    LOGGER.error("Cannot get compute margin for cnec {} in network variant.", cnecnetworkelementid);
+                }
```

Comment:
I think we can have the same error messages for both exception, with : 
   LOGGER.error(e.getMessage());
and 
   LOGGER.error("Cannot get compute margin for cnec {} in network variant.", cnecnetworkelementid);

Reply:
OK